framework/test/mocha/unit/controller/application_state.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
```

Comment:
shouldn't this be written in Jest?

Reply:
I can try to migrate it.

Reply:
I guess I can migrate it alongside I move the logic out of update function in #3141, What do you think?